### DISCUSSION TITLE
Cookies are always sent as an empty object

### DISCUSSION THREAD

**Discussion by skittlesaur** [trpc/trpc#2443] [Q&A]
I'm working on a tRPC project with Next.js and trying to create a middleware using request cookies. The problem is that whenever I use 'console.log(req.cookies)' in the 'createContext', it always outputs an empty object `{}`.

Also, for the frontend, the context in the 'withTRPC' function is always 'undefined', which might be the source of the problem?

For the frontend, I have this `_app.tsx` code:
```ts

const App: AppType = ({ Component, pageProps }: Props) => {
  return <Component {...pageProps} />
}

export default withTRPC<AppRouter>({
  config({ ctx }) {
    console.log(ctx) // <-- always outputs 'undefined'
    const url = 'http://localhost:5000/trpc'
    return {
      headers: () => {
        return {
            ...ctx?.req?.headers,
        }
      },
      url,
    }
  },
})(App)
```

Server context:
```ts
import { inferAsyncReturnType } from '@trpc/server'
import { CreateFastifyContextOptions } from '@trpc/server/adapters/fastify'
import { PrismaClient } from '@prisma/client'
import userToken from "lib/user-token"
 
const prisma: any = new PrismaClient()

export const createContext = async ({ req, res }: CreateFastifyContextOptions) => {
  console.log('cookies', req.cookies) // <-- always outputs 'cookies {}'
  return { req, res, prisma }
}

export type Context = inferAsyncReturnType<typeof createContext>
```

**Comment by JonParton** [comment]
Hmmn it feels too spookily similar not to be related to #2369... However it seems like you are already passing headers in the config of the withTRPC HOC ...  Maybe worth a closer look although note that is V10 I was working with!
